packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -577,7 +577,8 @@ class AndroidDevice extends Device {
       );
       // Package has been built, so we can get the updated application ID and
       // activity name from the .apk.
-      package = await AndroidApk.fromAndroidProject(project.android);
+      package = await ApplicationPackageFactory.instance
```

Comment:
eventually this should be passed as an argument or injected, but I did not want to refactor all of the device types at once

Reply:
SGTM